Consume libecl as targets

jokva opened the pull request:
This is a showcase of what downstream can look like with https://github.com/Statoil/libecl/pull/54 applied. Currently it's just the absolute minimal set of changes to demo what's going on.

Compiled and tested with `cmake -DCMAKE_MODULE_PATH=libecl/cmake/Modules -DCMAKE_MODULE_PATH=libecl/python/cmake/Modules`

pgdr commented on python/CMakeLists.txt:
```diff
@@ -1,29 +1,40 @@
+   set( PYTHON_PACKAGE_PATH "dist-packages")
+else()
+   set( PYTHON_PACKAGE_PATH "site-packages")
 endif()
```
this makes me a bit uneasy ... What is it that you _actually_ want to achieve here?

joakim-hove replied:
Well - this code has been around for  a long time, in this PR it is just moved up in the fil

http://stackoverflow.com/questions/42339034/python-module-in-dist-packages-vs-site-packages

http://stackoverflow.com/questions/9387928/whats-the-difference-between-dist-packages-and-site-packages

Could seem from the links that we should go to `site-packages` also on Debian derivates, however I vaguely seem to remember that `site-packages` was not in the default python search path on Debian/Ubuntu - might be mistake though.

pgdr replied:
Okay, nevermind, I guess it's okay.  I think it would potentially be better to deduce the information from Python, but I don't see a clean way for that either.

```python
import site
site.getsitepackages()
```

pgdr commented on python/CMakeLists.txt:
```diff
@@ -1,29 +1,40 @@
+set( PY_VERSION 2.7 )
+FIND_PACKAGE(PythonInterp ${PY_VERSION} EXACT)
+if (NOT DEFINED PYTHON_EXECUTABLE)
+   message("Python${PY_VERSION} interpreter not found - Python wrappers not enabled")
```
`message(WARNING "Python.....")` will result also in a warning to the user, which I think is more appropriate here; the user clearly wanted Python enabled.  Even an error would be appropriate, I think.

pgdr commented on python/CMakeLists.txt:
```diff
@@ -33,6 +44,14 @@ message(STATUS "Setting LD_LIBRARY_PATH: ${LD_PATH}")
 
+python_package(numpy 1.7.1)
+if (NOT DEFINED PY_numpy)
+   message("numpy module not found - Python wrappers not enabled")
```
Again a warning or an error would be better.  Actually, If libecl is built with Python support, then numpy is already found?  Pandas is also a requirement for building with python.

jokva committed:
Consume libecl as cmake targets

The upstream libecl exposes targets, not _LIBRARIES variables, and a
-config.cmake rather than a Find module.

jokva committed:
Fix miniconda download
